src/analysis/processing/qgsalgorithmdensifygeometriesbycount.cpp

```diff
@@ -0,0 +1,124 @@
+{
+  return QObject::tr( "This algorithm takes a polygon or line layer"
+                      "and generates a new one in which the geometries"
+                      "have a larger number of vertices than the "
+                      "original one. If the geometries have z or m values"
+                      "present then these will be linearly interpolated"
+                      "at the added nodes. The number of new vertices to"
+                      "add to each feature geometry is specified "
+                      "as an input parameter." );
+}
+
+QString QgsDensifyGeometriesByCountAlgorithm::shortDescription() const
+{
+  return QObject::tr( "Creates a densified version of geometries." );
+}
+
+QgsDensifyGeometriesByCountAlgorithm *QgsDensifyGeometriesByCountAlgorithm::createInstance() const
+{
+  return new QgsDensifyGeometriesByCountAlgorithm;
+
+}
+
+QList<int> QgsDensifyGeometriesByCountAlgorithm::inputLayerTypes() const
+{
+  return QList<int>() << QgsProcessing::TypeVectorLine << QgsProcessing::TypeVectorPolygon;
+}
+
+void QgsDensifyGeometriesByCountAlgorithm::initParameters( const QVariantMap &configuration )
+{
+  Q_UNUSED( configuration )
+  std::unique_ptr<QgsProcessingParameterNumber> verticesCnt = qgis::make_unique<QgsProcessingParameterNumber>( QStringLiteral( "VERTICES" ),
+      QObject::tr( "Number of vertices to add" ),
+      QgsProcessingParameterNumber::Integer,
+      1, false, 1, 10000000 );
+  verticesCnt->setIsDynamic( true );
+  verticesCnt->setDynamicPropertyDefinition( QgsPropertyDefinition( QStringLiteral( "VerticesCount" ), QObject::tr( "Vertices count" ), QgsPropertyDefinition::IntegerPositiveGreaterZero ) );
+  verticesCnt->setDynamicLayerParameterName( QStringLiteral( "INPUT" ) );
+  addParameter( verticesCnt.release() );
+}
+
+QString QgsDensifyGeometriesByCountAlgorithm::outputName() const
+{
+  return QObject::tr( "Densified" );
+}
+
+bool QgsDensifyGeometriesByCountAlgorithm::prepareAlgorithm( const QVariantMap &parameters, QgsProcessingContext &context, QgsProcessingFeedback *feedback )
+{
+  Q_UNUSED( feedback )
+  mVerticesCnt = parameterAsInt( parameters, QStringLiteral( "VERTICES" ), context );
+
+  mDynamicVerticesCnt = QgsProcessingParameters::isDynamic( parameters, QStringLiteral( "VERTICES" ) );
+  if ( mDynamicVerticesCnt )
+    mVerticesCntProperty = parameters.value( QStringLiteral( "VERTICES" ) ).value< QgsProperty >();
+
+  return true;
+}
+
+QgsFeatureList QgsDensifyGeometriesByCountAlgorithm::processFeature( const QgsFeature &feature, QgsProcessingContext &context, QgsProcessingFeedback *feedback )
+{
+  Q_UNUSED( context )
+  Q_UNUSED( feedback )
+  QgsFeature densifiedFeature = feature;
+
+  int verticesCnt = mVerticesCnt;
+  if ( mDynamicVerticesCnt )
+    verticesCnt = mVerticesCntProperty.valueAsInt( context.expressionContext(), verticesCnt );
+
+  if ( feature.hasGeometry() )
+    densifiedFeature.setGeometry( feature.geometry().densifyByCount( verticesCnt ) );
```

Comment:
I think we should put a safety check here of `if ( verticesCnt > 0 )`. It's possible a data defined parameter value may return a negative count... and I'm unsure if densifyByCount will handle that gracefully